packages/types/step/index.ts

```diff
@@ -0,0 +1,28 @@
+import {EventEmitter} from 'events';
+type StepEventHandler<T extends Step> = (e: StepEventData<T>) => any;
+type StepStatus = 'pending' | 'in progress' | 'done '; // can have more here
+export interface StepEventData<T extends Step> {
+  step: T;
+  date: Date;
+}
+interface Step extends EventEmitter {
```

Comment:
- Let's add Result interface as empty marker interface here.

- I'm not sure if using EventEmitter from node - this will mean that implementation here will be node.js specific. If we use plain interface then we can possibly have some parts of this code in client - which may be useful for Offline support.